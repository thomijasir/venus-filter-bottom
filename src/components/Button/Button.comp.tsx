import React, { FC, ReactNode } from 'react';
import { stringMatch } from '../../utils/Helper';
import './Button.scss';

export interface IProps {
  children: ReactNode;
  classStyle: string;
  onClick?: () => void;
  id?: string;
}

const Button: FC<IProps> = (props) => {
  const { children, classStyle, onClick, id } = props;
  const mainAction = (e: any) => {
    e.stopPropagation();
    e.preventDefault();
    if (onClick) {
      onClick();
    }
  };

  const isDisable = stringMatch(classStyle, 'inactive') !== null;

  return (
    <button
      id={id}
      type="button"
      onClick={mainAction}
      className={`button ${classStyle}`}
      disabled={isDisable}
    >
      {children}
    </button>
  );
};

export default Button;
