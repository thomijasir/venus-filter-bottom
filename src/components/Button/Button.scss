@import '../../assets//style/mixin';
@import '../../assets/style/colors';
@import '../../assets//style/var';
.button {
  border-width: 1px;
  border-style: solid;
  padding: 8px 16px;
  border-radius: 0.05rem;
  outline: 0;
  background: $white;
  font-family: $font-second;
  font-size: 16px;
  text-decoration: none;
  border: 1px solid $shadow-blue;
  cursor: pointer;
  @include xSmall() {
    padding: 0.06rem;
  }
  &.uppercase {
    text-transform: uppercase;
  }
  &.primary {
    color: $white;
    background: $shadow-blue;
    &.inactive {
      color: $white;
      background: $very-light-two;
      border: 1px solid $very-light-two;
    }
  }
  &.invert {
    color: $shadow-blue;
    background: $white;
  }
  &.outline {
    color: $shadow-blue;
    &.inactive {
      color: $very-light-two;
      border: 1px solid $very-light-two;
    }
  }
  &.full {
    width: 100%;
  }
  &.no-margin {
    margin: 0rem;
  }
  &.bold {
    font-weight: 600;
  }
  &.small {
    font-size: 0.12rem;
  }
  &.big {
    font-size: 16px;
    font-weight: 600;
    min-height: 0.45rem;
    @include xSmall() {
      font-size: 0.15;
    }
  }
  &.nowrap {
    white-space: nowrap;
  }
  &.denom {
    font-size: 0.25rem;
    width: 0.37rem;
    height: 0.2rem;
    padding-bottom: 0.35rem;
    padding-top: 0;
  }
}
