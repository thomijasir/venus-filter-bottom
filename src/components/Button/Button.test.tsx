import React from 'react';
import render from 'react-test-renderer';
import Button from './Button.comp';

afterEach(() => {
  jest.clearAllMocks();
});

describe('Error General Component Snap Test', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const compProps = {
        classStyle: 'primary',
        onClick: () => {},
        id: 'btn',
      };
      const wrapper = render.create(<Button {...compProps}>Sample</Button>);
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });
});

// import React from 'react';
// import { shallow } from 'enzyme';
// import toJson from 'enzyme-to-json';
// import Button from './Button.comp';

// const event = { preventDefault: () => {}, stopPropagation: () => {} };
// let wrapper;
// let instance;

// const props = {
//   style: 'outline',
//   onClick: jest.fn(),
// };

// jest.useFakeTimers();

// beforeEach(() => {
//   wrapper = shallow(<Button {...props} />);
//   instance = wrapper.instance();
//   jest.spyOn(event, 'preventDefault');
//   jest.spyOn(event, 'stopPropagation');
// });

// afterEach(() => {
//   jest.clearAllMocks();
// });

// describe('BillersPage Snap Test', () => {
//   describe('render()', () => {
//     test('renders with success result', () => {
//       expect(toJson(wrapper)).toMatchSnapshot();
//     });
//     test('Add class renders with success result', () => {
//       const nProps = {
//         style: 'outline',
//         onClick: jest.fn(),
//         addClass: 'inactive',
//       };
//       const nWrapper = shallow(<Button {...nProps} />);
//       expect(toJson(nWrapper)).toMatchSnapshot();
//     });
//   });
//   describe('strRegExPattern()', () => {
//     test('get match pattern strRegExPattern', () => {
//       const result = instance.stringMatch('data inactive', 'inactive');
//       expect(result.length).toEqual(1);
//     });
//   });
//   describe('mainAction()', () => {
//     test('mainAction excuted', () => {
//       instance.mainAction(event);
//       expect(event.preventDefault).toHaveBeenCalled();
//       expect(event.stopPropagation).toHaveBeenCalled();
//       expect(props.onClick).toHaveBeenCalled();
//     });
//   });
// });
