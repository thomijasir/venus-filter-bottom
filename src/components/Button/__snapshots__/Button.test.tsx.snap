// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error General Component Snap Test render() renders the component 1`] = `
<button
  className="button primary"
  disabled={false}
  id="btn"
  onClick={[Function]}
  type="button"
>
  Sample
</button>
`;
